packages/cms-lib/lib/uploadFolder.js

```diff
@@ -41,38 +70,45 @@ async function uploadFolder(portalId, src, dest, { mode, cwd }) {
     })
     .filter(createIgnoreFilter(cwd));
 
+  const filesByType = getFilesByType(allowedFiles);
+
   const failures = [];
-  await queue.addAll(
-    filesToUpload.map(file => {
-      const relativePath = file.replace(regex, '');
-      const destPath = convertToUnixPath(path.join(dest, relativePath));
-      return async () => {
-        logger.debug('Attempting to upload file "%s" to "%s"', file, destPath);
-        try {
-          await upload(portalId, file, destPath, apiOptions);
-          logger.log('Uploaded file "%s" to "%s"', file, destPath);
-        } catch (error) {
-          if (isFatalError(error)) {
-            throw error;
-          }
-          logger.debug(
-            'Uploading file "%s" to "%s" failed so scheduled retry',
-            file,
-            destPath
-          );
-          if (error.response && error.response.body) {
-            logger.debug(error.response.body);
-          } else {
-            logger.debug(error.message);
-          }
-          failures.push({
-            file,
-            destPath,
-          });
+
+  const uploadFile = file => {
+    const relativePath = file.replace(regex, '');
+    const destPath = convertToUnixPath(path.join(dest, relativePath));
+    return async () => {
+      logger.debug('Attempting to upload file "%s" to "%s"', file, destPath);
+      try {
+        await upload(portalId, file, destPath, apiOptions);
+        logger.log('Uploaded file "%s" to "%s"', file, destPath);
+      } catch (error) {
+        if (isFatalError(error)) {
+          throw error;
         }
-      };
-    })
-  );
+        logger.debug(
+          'Uploading file "%s" to "%s" failed so scheduled retry',
+          file,
+          destPath
+        );
+        if (error.response && error.response.body) {
+          logger.debug(error.response.body);
+        } else {
+          logger.debug(error.message);
+        }
+        failures.push({
+          file,
+          destPath,
+        });
+      }
+    };
+  };
+
+  for (let i = 0; i < filesByType.length; i++) {
```

Comment:
fyi, the old school for loop is necessary due to async/await.

Reply:
do you think it's worth adding this comment to the code?

Reply:
a comment wouldn't hurt so added it.